plugins/plugin-traefik/plugin-traefik-docker/src/main/java/org/eclipse/che/plugin/traefik/TraefikCreateContainerInterceptor.java

```diff
@@ -0,0 +1,148 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.traefik;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.name.Named;
+
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.eclipse.che.commons.annotation.Nullable;
+import org.eclipse.che.plugin.docker.client.DockerConnector;
+import org.eclipse.che.plugin.docker.client.json.ContainerConfig;
+import org.eclipse.che.plugin.docker.client.json.ImageInfo;
+import org.eclipse.che.plugin.docker.client.params.CreateContainerParams;
+import org.eclipse.che.plugin.docker.client.params.InspectImageParams;
+import org.eclipse.che.plugin.docker.machine.CustomServerEvaluationStrategy;
+import org.eclipse.che.plugin.docker.machine.ServerEvaluationStrategy;
+import org.eclipse.che.plugin.docker.machine.ServerEvaluationStrategyProvider;
+
+import javax.inject.Inject;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static java.lang.String.format;
+
+/**
+ * Intercept the calls on createContainer on docker Connector.
+ *
+ * @author Florent Benoit
+ */
+public class TraefikCreateContainerInterceptor implements MethodInterceptor {
+
+    /**
+     * Inject the server evaluation strategy provider.
+     */
+    private ServerEvaluationStrategyProvider serverEvaluationStrategyProvider;
+
+    /**
+     * Template.
+     */
+    private String template;
+
+
+    /**
+     * Grab labels of the config and from image to get all exposed ports and the labels defined if any
+     *
+     * @param methodInvocation
+     *         intercepting data of createContainer method on {@link DockerConnector}
+     * @return the result of the intercepted method
+     * @throws Throwable
+     *         if there is an exception
+     */
+    @Override
+    public Object invoke(MethodInvocation methodInvocation) throws Throwable {
+        ServerEvaluationStrategy serverEvaluationStrategy = serverEvaluationStrategyProvider.get();
+        // Abort if custom server evaluation strategy is not enabled.
+        if (!(CustomServerEvaluationStrategy.class.isInstance(serverEvaluationStrategy))) {
+            return methodInvocation.proceed();
+        }
+        final CustomServerEvaluationStrategy customServerEvaluationStrategy = (CustomServerEvaluationStrategy)serverEvaluationStrategy;
+
+        // Get the connector
+        DockerConnector dockerConnector = (DockerConnector)methodInvocation.getThis();
+
+        // only one parameter which is CreateContainerParams
+        CreateContainerParams createContainerParams = (CreateContainerParams)methodInvocation.getArguments()[0];
```

Comment:
well due to all changes on https://github.com/eclipse/che/pull/5052 I will check once it is merged but either we can drop support for OpenShift connector or if there are issues, it will be fixed.